### DISCUSSION TITLE
smithy4s MQTT in scope?

### DISCUSSION THREAD

**Discussion by alexcardell** [disneystreaming/smithy4s#726] [General]
Is allowing `smithy.mqtt` traits considered in scope for smithy4s? 

I've had a go at this myself by adding the `smithy-mqtt-traits` package to the codegen module, and added `smithy.mqtt` to allowed namespaces where seems reasonable, but no luck! 

The use case here is I'd like to write a backend for SQS and RabbitMQ (using ITV/bucky)

**Comment by kubukoz** [comment]
Can you share your attempt's code? I can take a look. My opinion would be that we should have a new module with this, like `aws-kernel` for AWS traits. I would absolutely love to see an implementation for RabbitMQ.

**Reply by alexcardell** [reply]
https://github.com/disneystreaming/smithy4s/pull/727/files

Pointing out that module helped me fix my build config and the code generation is working in another project! Thanks so much

**Reply by kubukoz** [reply]
right, can you show what you've tried in another project to generate these traits? I thought `allowedNamespaces` would work for plugin users.

**Reply by kubukoz** [reply]
In general, if we're not going to implement anything on top in this repo, the best way would be to get you a standalone setup - adding a tiny module in here may not feel like much, but we'd have to cross-compile that several times, and the build matrix is already huge.

**Reply by alexcardell** [reply]
I'm an idiot and missed the *but* in `By default, Smithy4s tries to generate everything but shapes that are in the following namespaces:`

Got it working now, thanks for your help, I'll update this thread once I have a rabbit client

**Reply by kubukoz** [reply]
Happens to all of us ;) perhaps that sentence could be reworded to be less confusing.

**Reply by Baccata** [reply]
> perhaps that sentence could be reworded to be less confusing. 

yup, happy for someone to change it to something more concise / clear
